### PR TITLE
chore(MeshHTTPRoute): support url rewriting

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? -- #5470 
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests -- none for this feature
- [x] Manual Universal Tests --
- [ ] Manual Kubernetes Tests -- none yet
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshhttproute/plugin/v1alpha1/plugin_test.go [hunk 1]
```diff
@@ -407,6 +407,65 @@ var _ = Describe("MeshHTTPRoute", func() {
...
+				Weight: 1,
+			}},
+		}
+		return outboundsTestCase{
+			xdsContext: xds_context.Context{
+				ControlPlane: &xds_context.ControlPlaneContext{
+					Secrets: &xds.TestSecrets{},
+				},
+				Mesh: xds_context.MeshContext{
+					Resource:    builders.Mesh().WithName("default").Build(),
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshhttproute/plugin/v1alpha1/plugin_test.go:425]: nit^2: `samples.MeshDefault()`

File: pkg/plugins/policies/meshhttproute/xds/configurer.go [hunk 2]
```diff
@@ -21,27 +21,50 @@ type RoutesConfigurer struct {
...
+	*envoy_route.RouteMatch
+}
+
+func (m withPrefixMatch) getRouteMatch() *envoy_route.RouteMatch { return m.RouteMatch }
+
+type withoutPrefixMatch struct {
+	*envoy_route.RouteMatch
+}
+
+func (m withoutPrefixMatch) getRouteMatch() *envoy_route.RouteMatch { return m.RouteMatch }
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshhttproute/xds/configurer.go:61]: Maybe I'm missing something, but if all implementations are the same, wouldn't it be better to have something like
```
type match struct{
  routeMatch *envoy_route.RouteMatch
  prefixMatch bool
}
```
?

File: pkg/plugins/policies/meshhttproute/xds/filters.go [hunk 3]
```diff
@@ -105,3 +108,47 @@ func requestRedirect(redirect api.RequestRedirect, envoyRoute *envoy_route.Route
...
+						Pattern: &envoy_type_matcher.RegexMatcher{
+							EngineType: &envoy_type_matcher.RegexMatcher_GoogleRe2{
+								GoogleRe2: &envoy_type_matcher.RegexMatcher_GoogleRE2{},
+							},
+							Regex: `.*`,
+						},
+						Substitution: *rewrite.Path.ReplacePrefixMatch,
+					}
+				} else if envoyRoute.Match.GetPrefix() != "" {
+					// We have the "predix /prefix/" match case
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshhttproute/xds/filters.go:145]: nit: typo

File: pkg/plugins/policies/meshhttproute/xds/configurer.go (hunk 2 above)
Comment (michaelbeaumont) [reply pkg/plugins/policies/meshhttproute/xds/configurer.go:61]: yes, got too hung up on using a sum type which are very painful in go.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**